src/main/java/io/asyncer/r2dbc/mysql/MySqlConnection.java

```diff
@@ -303,13 +303,25 @@ public IsolationLevel getTransactionIsolationLevel() {
         return currentLevel;
     }
 
+    /**
+     * Gets session transaction isolation level(Only for testing).
+     *
+     * @return session transaction isolation level.
+     */
+    IsolationLevel getSessionTransactionIsolationLevel() {
+        return sessionLevel;
+    }
+
     @Override
     public Mono<Void> setTransactionIsolationLevel(IsolationLevel isolationLevel) {
         requireNonNull(isolationLevel, "isolationLevel must not be null");
 
         // Set next transaction isolation level.
         return QueryFlow.executeVoid(client, "SET TRANSACTION ISOLATION LEVEL " + isolationLevel.asSql())
-            .doOnSuccess(ignored -> setIsolationLevel(isolationLevel));
+            .doOnSuccess(ignored -> {
+                this.sessionLevel = isolationLevel;
+                resetIsolationLevel();
```

Comment:
Can modify the current isolation level only when there is no ongoing transaction. In MySQL, it is not permitted to change the isolation level of a transaction that is already in progress.

To accomplish this functionality, I suggest referring to `AbstractTransactionState`.

Reply:
You are right. The current isolation level will be reset automatically when commiting or rollbacking, please see https://github.com/asyncer-io/r2dbc-mysql/blob/96c11dcfe86e9a95cc857a5bfadc7906be5ef953/src/main/java/io/asyncer/r2dbc/mysql/QueryFlow.java#L1088